kong/plugins/key-auth/api.lua

```diff
@@ -52,5 +52,10 @@ return {
     DELETE = function(self, dao_factory)
       crud.delete(self.keyauth_credential, dao_factory.keyauth_credentials)
     end
+  },
+  ["/plugins/key-auth/consumers"] = {
```

Comment:
It seems to me like the intent of such an endpoint is to retrieve a paginated list key-auth credentials. One of the use-cases that this solves is "retrieving a particular key-auth in order to read it's consumer_id and retrieve the Consumer associated with a key-auth credential" (as the title of this PR suggests). I think this would mean that users need to make the following request:

```
/plugins/key-auth/consumers?id=...
or
/plugins/key-auth/consumers?key=...
``` 

However, such a request could only return a single result, as a key-auth credential can only be associated with a single Consumer. Additionally, the current implementation seems to return key-auth credential, and makes forces the client to make a subsequent request to retrieve the full Consumer entity. Finally, the URL of this request is confusing in the sense that user isn't sure if the response will be a list of Plugins, key-auth, or Consumers...

If we want to add both a paginated set of all key-auth credentials, and the ability to retrieve the Consumer associated with a credential, it might be friendlier and more appropriate (RESTful) to solve those two requirements with two endpoints:
```
/key-auth/
/consumers/key-auth/:credential_key_or_id
```

The reasoning is that in our RESTful interface, we try to prefix a resource's endpoint with the type of entities the endpoint will return. The first endpoint returns a list of key-auth credentials. The second one directly returns a single Consumer associated with this credential (or `404 Not Found`) without the need for the client to send 2 requests (the first one for the key-auth credential, and the next one to finally request the Consumer from the retrieved `consumer_id`).

What do you think?